Bump version and amend notes to include latest PRs (#4634)
miekg/coredns: notes/coredns-1.8.4.md

description = "CoreDNS-1.8.4 Release Notes."
tags = ["Release", "1.8.4", "Notes"]
release = "1.8.4"
date = 2021-05-18T07:00:00+00:00
author = "coredns"
+++

The CoreDNS team has released
[CoreDNS-1.8.4](https://github.com/coredns/coredns/releases/tag/v1.8.4). This release includes a
bunch of bugfixes and a few enhancements mostly in the *dnssec* and *kubernetes* plugins, and a new
(small) plugin called *minimal*.

> It also include a fix when using the "reverse zone syntax", i.e. 10.0.0.0/15, now return the proper

Reviewer: ```suggestion
It also include a fix when using the "reverse zone cidr syntax", e.g. 10.0.0.0/15, now return the proper
```

> It also include a fix when using the "reverse zone syntax", i.e. 10.0.0.0/15, now return the proper

Reviewer: Should highlight required changes to existing external plugins introduce with the plugin level reverse zone changes.

set of reverse zones.

> 

Reviewer: ```suggestion
**NOTE:** This release deprecates `Host.Normalize()` and `Host.MustNormalize()`. For external plugins that use these functions, see https://github.com/coredns/coredns/pull/4621 for examples of how to update.
```

## Noteworthy Changes

> 

Reviewer 2: Extra newline?

* core: fix reverse zones expansion (https://github.com/coredns/coredns/pull/4538)
* plugins: fix Normalize (https://github.com/coredns/coredns/pull/4621)

> * plugin/bind: Bind by interface name (https://github.com/coredns/coredns/pull/4522)

Reviewer: ```suggestion
* reverse zone: make Normalize return proper reverse zones (https://github.com/coredns/coredns/pull/4621)
* plugin/bind: Bind by interface name (https://github.com/coredns/coredns/pull/4522)
```

* plugin/bind: Exclude interface or ip address  (https://github.com/coredns/coredns/pull/4543)

> * plugin/dnssec: heck for two days of remaining validity (https://github.com/coredns/coredns/pull/4606)

Reviewer: ```suggestion
* plugin/dnssec: Check for two days of remaining validity (https://github.com/coredns/coredns/pull/4606)
```

* plugin/dnssec: interface type correction for `periodicClean` sig validity check (https://github.com/coredns/coredns/pull/4608)
* plugin/dnssec: use entire RRset as key input (https://github.com/coredns/coredns/pull/4537)

> * plugin/etcd: Fix etcd to use v3.5.0-alpha.0 (https://github.com/coredns/coredns/pull/4628)

Reviewer: ~v3.5.0-beta.1~ v3.5.0-beta.2 just got released.

> * plugin/etcd: Fix etcd to use v3.5.0-alpha.0 (https://github.com/coredns/coredns/pull/4628)

Reviewer: ```suggestion
* plugin/etcd: Bump etcd to v3.5.0-beta.3 (https://github.com/coredns/coredns/pull/4638)
```
